pkg/k8s/apis/cilium.io/v2/lrp_types.go

```diff
@@ -0,0 +1,146 @@
+//  Copyright 2020 Authors of Cilium
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package v2
+
+import (
+	"github.com/cilium/cilium/pkg/policy/api"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:singular="localredirectpolicy",path="localredirectpolicies",scope="Namespaced",shortName={lrp}
+// +kubebuilder:printcolumn:JSONPath=".metadata.creationTimestamp",name="Age",type=date
+
+// LocalRedirectPolicy is a Kubernetes Custom Resource that contains a
+// specification to redirect traffic locally within a node.
+type LocalRedirectPolicy struct {
+	// +k8s:openapi-gen=false
+	// +deepequal-gen=false
+	metav1.TypeMeta `json:",inline"`
+	// +k8s:openapi-gen=false
+	// +deepequal-gen=false
+	metav1.ObjectMeta `json:"metadata"`
+
+	// Specification of the desired behavior of the local redirect policy
+	Spec LocalRedirectPolicySpec `json:"spec,omitempty"`
+
+	// Most recent status of the local redirect policy
+	// Read-only
+	//
+	// +deepequal-gen=false
+	// +kubebuilder:validation:Optional
+	Status LocalRedirectPolicyStatus `json:"status"`
+}
+
+type L3L4Info struct {
+	// ToIP is a destination IP address for traffic to be redirected.
+	//
+	// Example:
+	// Any traffic destined to '169.254.169.254' is redirected.
+	//
+	// +kubebuilder:validation:Optional
+	ToIP string `json:"toIP,omitempty"`
+
+	// ToPort is a destination port for traffic to be redirected.
+	//
+	// Example:
+	// Any traffic destined to port '53' is redirected.
+	//
+	// +kubebuilder:validation:Optional
+	ToPort string `json:"toPort,omitempty"`
+
+	// L4Proto is layer 4 protocol number for traffic to be redirected.
+	//
+	// +kubebuilder:validation:Enum=TCP;UDP
+	L4Proto api.L4Proto `json:"L4Proto,omitempty"`
+}
+
+// RedirectFrom is a frontend configuration that determines traffic that needs to be redirected.
+// The configuration must be specified using an L3L4 tuple or a Kubernetes service.
+type RedirectFrom struct {
+	// ToL3L4Info is a destination tuple {IP, port, protocol} that identifies traffic
+	// to be redirected.
+	//
+	// +kubebuilder:validation:Optional
+	ToL3L4 L3L4Info `json:"L3L4Info,omitempty"`
+
+	// ToService is a destination Kubernetes service that identifies traffic
+	// to be redirected. The service namespace must match the namespace of the
+	// parent Local Redirect Policy.
```

Comment:
This description will be shared for ClusterLRP and LRP so it would make sense to have documentation for the CLRP case where we can specify any service namespace in a CLRP.